balancer_wrapper.go

```diff
@@ -436,17 +458,59 @@ func (acbw *acBalancerWrapper) RegisterHealthListener(listener func(balancer.Sub
 		return
 	}
 
+	// Client side health checking is enabled when all the following
+	// conditions are satisfied:
+	// 1. Health checking is not disabled using the dial option.
+	// 2. The health package is imported.
+	// 3. The health check config is present in the service config.
+	healthCheckEnabled := !acbw.ccb.cc.dopts.disableHealthCheck
+	regHealthLisFn := internal.RegisterClientHealthCheckListener
+	if regHealthLisFn == nil {
+		// The health package is not imported.
+		healthCheckEnabled = false
+	}
+	var cfg *healthCheckConfig
+	if healthCheckEnabled {
+		// Avoid acquiring cc.mu unless necessary.
+		cfg = acbw.ac.cc.healthCheckConfig()
+		healthCheckEnabled = cfg != nil
+	}
```

Comment:
What do you feel about making a helper method on `acBalancerWrapper` that returns `(string, bool)` to indicate whether health checking is enabled or not (and the first return value will contain the service name when the second value is `true`).

The reason why a helper would be nicer is because we wont have to successively rewrite a local variable as is done currently, and instead can return `false` whenever one of the conditions is not met.

Reply:
That's a nice idea. I made the helper returns the `regHealthLisFn` too. Since the `regHealthLisFn` must be non-nil when health checking is enabled, I didn't removed the bool from the return params.